modules/dxp/apps/portal-workflow/portal-workflow-metrics-web/src/main/resources/META-INF/resources/js/components/instance-list-page/InstanceListPageItem.es.js

```diff
@@ -152,16 +195,46 @@ function Item({totalCount, ...instance}) {
 			<ClayTable.Cell>
 				<div
 					className={`due-date ${
-						slaRunning ? slaStatusIconInfo.textColor : 'text-info'
+						criticalSlaResults?.length
+							? slaStatusIconInfo.textColor
+							: 'text-info'
 					}`}
 				>
-					{!slaResult || !slaRunning ? (
+					{!criticalSlaResults?.length ? (
 						'-'
 					) : (
-						<>
-							<span className="due-date-badge"></span>
-							{dateOverdue}
-						</>
+						<ClayPopover
```

Comment:
Hey @jotascript and @naraandrade What do you think if we extract the popover logic for a new Component?

Thx

Reply:
@rafaprax 
Are yousuggesting that all the code inside the ClayPopover have to be a new component? then we have to import and use it? if is it, I think this is a great idea, thi way this way can be more readable

Reply:
@ramonmsp It is exactly what I meant

Reply:
@rafaprax and @ramonmsp, I think we can keep it that way because this is the only place that it is being used so far. We can change this in the future if necessary.

Reply:
Hey @jotascript,

I can see your point of view but in this case I was thinking more about readability than reusability. And from now on we have the logic of building a table and building the popover mixed up.